src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,190 @@
+        FileWriter fw = null;
+        try {
+            fw = new FileWriter(dataPath);
+            String textToAdd = Task.getData();
+            System.out.println(textToAdd);
+            fw.write(textToAdd);
+            fw.close();
+            System.out.println("Saved successfully.");
+            //if the file does not exist
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public static void loadData(String dataPath){
+        File f = new File(dataPath);
+        System.out.println("full path: " + f.getAbsolutePath());
+        try {
+            Scanner s = new Scanner(f);
+            while (s.hasNext()){
+                loadTask(s.nextLine());
+
+            }
+            //if the file does not exist
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        }
+
+    }
+
+    public static void loadTask(String taskText){
+        String[] arrOfText = taskText.split("\\|");
+        if (arrOfText.length == 3) {
+            Todo todo = new Todo(arrOfText[2]);
+            if (Integer.parseInt(arrOfText[1]) == 1){
+                todo.markAsDone();
+            }
+            Task.addTask(todo);
+        }
+        else if (arrOfText.length == 4) {
+            switch (arrOfText[0]){
+            case "D":
+                Deadline deadline = new Deadline(arrOfText[2],arrOfText[3]);
+                if (Integer.parseInt(arrOfText[1]) == 1){
+                    deadline.markAsDone();
+                }
+                Task.addTask(deadline);
+                break;
+            case "E":
+                Event event = new Event(arrOfText[2],arrOfText[3]);
+                if (Integer.parseInt(arrOfText[1]) == 1){
+                    event.markAsDone();
+                }
+                Task.addTask(event);
+                break;
+            default:
+                System.out.println("Invalid case.");
+                break;
+            }
+        }
+        else {
+            System.out.println("No task left.");
+        }
+    }
+
+    public static void getCommand(String input) throws IllegalCommandException, NonExistentCommandException{
+        String[] arrOfInput = input.split(" ",2);
+        switch (arrOfInput[0]) {
+        case "bye":
+            Command.exit();
+            break;
+        case "list":
+            Command.list();
+            break;
+        case "todo":
+            if (arrOfInput.length < 2){
+                throw new IllegalCommandException();
+            }
+
+            Command.addTodo(arrOfInput[1]);
+            break;
+        case "deadline":
+            if (arrOfInput.length < 2){
+                throw new IllegalCommandException();
+            }
+            try {
+                Command.addDDL(arrOfInput[1]);
+            }
+            catch (IllegalFormatException e){
+                System.out.println("OOPS!!! The format is not correct.");
+            }
+
+            break;
```

Comment:
Avoid long methods (>30 LoC). Perhaps you could refactor the logic in the case statements to separate methods?